### PR TITLE
Tic Tac Toe Javascript

### PR DESCRIPTION
**Pull request by fivan18**
# Completed project - Tic Tac Toe Javascript

This project consisted of making a Tic Tac Toe game you can play in your browser. 

## Project specific requirements:

- [ ] Set up your project with a HTML, CSS and Javascript files
- [ ] Store the gameboard as an array inside of a Gameboard object
- [ ] Set up your HTML and write a JavaScript function that will render the contents of the gameboard array
- [ ] Build the functions that allow players to add marks to a specific spot on the board, and then tie it to the DOM, letting players click on the gameboard to place their marker
- [ ] Build the logic that checks for when the game is over
- [ ] Clean up the interface to allow players to put in their names, include a button to start/restart the game and add a display element that congratulates the winning player
- [ ] If you’re feeling ambitious create an AI so that a player can play against the computer

## Authors

👤 **Ivan Ulises Guzman Sanchez**
- Github: [@fivan18](https://github.com/fivan18)

👤 **Juan Pablo Gil**
- [GitHub](https://github.com/JuanPabloGil )

### CODE REVIEW

File: scripts/game_controller.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  Player(
+    playerTwo,
+    '✘',
+  )];
+
+  const currentPlayer = 0;
+
+  const displayOnTitleMessage = function displayOnTitleMessage(message) {
+    doc.getElementById('cPlayer').textContent = message;
+  };
```
Comment (mcrd25) [inline scripts/game_controller.js:25]: Don't forget we ought to separate DOM manipulation from application logic! 🙂 

See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) from pathwright.

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required 🔍 👀 

Hi @fivan18! Hi @JuanPabloGil! 👋 

Great job! Especially with your AI feature! Nice touch!

Address my review comments appropriately and you should get the 👍  for the next review!

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

File: scripts/game_controller.js (hunk 1 above)
Comment (fivan18) [reply scripts/game_controller.js:25]: Hi Maya,
 First of all, thanks for the review.

The [gameController](https://github.com/JuanPabloGil/Tic-Tac-Toe-JS/blob/game/scripts/game_controller.js) module doesn't contain the application logic. This module contains methods to manipulate the DOM: `render` and `displayOnTitleMessage `
https://github.com/JuanPabloGil/Tic-Tac-Toe-JS/blob/b196e9001d92ec8a295a897c1cf6328b63b639fa/scripts/game_controller.js#L3-L8

https://github.com/JuanPabloGil/Tic-Tac-Toe-JS/blob/b196e9001d92ec8a295a897c1cf6328b63b639fa/scripts/game_controller.js#L23-L25

It also serves to store the `players`,  `currentPlayer` and if the user is playing `againstComputer `:

https://github.com/JuanPabloGil/Tic-Tac-Toe-JS/blob/b196e9001d92ec8a295a897c1cf6328b63b639fa/scripts/game_controller.js#L10-L19

https://github.com/JuanPabloGil/Tic-Tac-Toe-JS/blob/b196e9001d92ec8a295a897c1cf6328b63b639fa/scripts/game_controller.js#L21

https://github.com/JuanPabloGil/Tic-Tac-Toe-JS/blob/b196e9001d92ec8a295a897c1cf6328b63b639fa/scripts/game_controller.js#L27

In the [main.js](https://github.com/JuanPabloGil/Tic-Tac-Toe-JS/blob/game/scripts/main.js) script we set the handlers for each element the user will interact with. Those handlers contain the application logic.

Comment (mcrd25) [reply scripts/game_controller.js:25]: Hey, Sorry about this! The game controller ought to have application logic or a similar Game object. I must have misunderstood your intentions

Comment (mcrd25) [reply scripts/game_controller.js:25]: But this is fine, You may submit another review and you should get the 👍

Comment (fivan18) [reply scripts/game_controller.js:25]: I've requested for a new code review. Thanks.

**Review by mcrd25 — APPROVED** [review]
# Changes Approved 💪 ㊗️ 🎉 

Hi Team!

It's time for blastoff 🚀 

You may now merge to `dev` and `master`. 🙂

**Pull request merged**